lxd/migration/migration_volumes.go

```diff
@@ -205,7 +205,9 @@ func MatchTypes(offer *MigrationHeader, fallbackType MigrationFSType, ourTypes [
 				}
 			}
 
-			if offer.Refresh != nil && *offer.Refresh {
+			// Only check for optimized refresh features if refreshing.
+			// It should also be considered that offer.Refresh can be nil for older LXD versions.
+			if offer.GetRefresh() || (refresh && offer.Refresh == nil) {
```

Comment:
I trust this works, but I'm still finding it difficult to understand why if offer.Refresh is nil why we would still potentially try and do optimized refresh mode (if the offer/response header supports it)?

Reply:
I checked this again.

When copying/refreshing 5.0 -> 5.0, `offer.Refresh` is always `nil`. That's because the source/sender doesn't set the field. The target lets the source know whether or not it's a refresh. If we removed this check, it would start using rsync.

Reply:
Ok so its for lxd 5.0.0, which had partial (and fairly buggy) optimized refresh support. I wonder if we shouldn't just disable optimized refresh in that case and wait for your fixes to be added to lxd 5.0.1. This will be simpler.

Also you mention lxd 5.0 to lxd 5.0, but that can't be changed. Only lxd 5.0 to lxd edge can be changed now right? So am I missing something?

Reply:
Oh right, I see. Wouldn't it be simpler and more consistent to have the source also set that into the offer header, that way you wouldn't need the extra argument on this function.

Reply:
That way we would just check for GetRefresh()

Reply:
> Wouldn't it be simpler and more consistent to have the source also set that into the offer header, that way you wouldn't need the extra argument on this function.

If I'm reading the code correctly, that would mean an API change to `api.InstancePost` as that doesn't contain a `Refresh` field. The `newMigrationSource` function would then need to take that value.

I don't think we should do an API change just for this.

Reply:
No agreed. I think you can just pass c.refresh into offerHeader.Refresh?

Reply:
> I think you can just pass c.refresh into offerHeader.Refresh?

I can confirm that this works. I'll update the PR. We can then also get rid of the `refresh` argument in `MatchTypes`.

Reply:
Thanks